duecredit/entries.py

```diff
@@ -7,23 +7,24 @@
 #   under MIT license
 #
 # ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
-import re
-from typing import Optional
+from __future__ import annotations
 
 import logging
+import re
+
 lgr = logging.getLogger('duecredit.entries')
 
 
 class DueCreditEntry:
     _key : str
 
-    def __init__(self, rawentry:str, key: Optional[str] = None) -> None:
+    def __init__(self, rawentry:str, key: str | None = None) -> None:
         self._rawentry = rawentry
         self._key = key or rawentry.lower()
 
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, DueCreditEntry):
-            raise NotImplemented
+            return NotImplemented
```

Comment:
I don't think it makes sense to return here

Reply:
It does make sense: https://docs.python.org/3/library/constants.html#NotImplemented

Reply:
ah -- cool, learned something new again, should have checked.

<details>
<summary>FWIW - here is a trial on a simple case</summary> 

```shell
❯ python -m trace --trace notimpl.py
 --- modulename: notimpl, funcname: <module>
notimpl.py(1): class A:
 --- modulename: notimpl, funcname: A
notimpl.py(1): class A:
notimpl.py(2):     def __eq__(self, other):
notimpl.py(8): print(A() == A())
 --- modulename: notimpl, funcname: __eq__
notimpl.py(3):         if isinstance(other, A):
notimpl.py(4):             return True
True
notimpl.py(9): print(A() == 1)
 --- modulename: notimpl, funcname: __eq__
notimpl.py(3):         if isinstance(other, A):
notimpl.py(6):             return NotImplemented
False
notimpl.py(10): print(A() != 1)
 --- modulename: notimpl, funcname: __eq__
notimpl.py(3):         if isinstance(other, A):
notimpl.py(6):             return NotImplemented
True
❯ cat notimpl.py
class A:
    def __eq__(self, other):
        if isinstance(other, A):
            return True
        else:
            return NotImplemented

print(A() == A())
print(A() == 1)
print(A() != 1)
```
</details>

I will revert that f470c3e8c41a02f70f0a700d81e7f1b2533abb60 I merged in rush, directly in master